### PR TITLE
[14.0] shopinvader_search_engine: support lang agnostic indexes

### PR DESCRIPTION
**Pull request by simahawk**
- [x] Depends on https://github.com/OCA/search-engine/pull/87

### CODE REVIEW

File: shopinvader_search_engine/models/se_index.py [hunk 1]
```diff
@@ -11,11 +11,14 @@ class SeIndex(models.Model):
...
+            backend = backend.browse(backend_id)
         for rec in self:
-            active_id = self.env.context.get("shopinvader_backend_id")
-            active_id = self.env["shopinvader.backend"].browse(active_id)
-            if active_id and rec.lang_id in active_id.lang_ids:
+            if not rec.lang_id or not backend:
                 rec.is_valid = True
-            else:
-                rec.is_valid = False
+                continue
```
Comment (hparfr) [inline shopinvader_search_engine/models/se_index.py:23]: `else:` ?

File: shopinvader_search_engine/models/se_index.py [hunk 2]
```diff
@@ -11,11 +11,14 @@ class SeIndex(models.Model):
...
+        backend = self.env["shopinvader.backend"]
+        backend_id = self.env.context.get("shopinvader_backend_id")
+        if backend_id:
+            backend = backend.browse(backend_id)
         for rec in self:
-            active_id = self.env.context.get("shopinvader_backend_id")
-            active_id = self.env["shopinvader.backend"].browse(active_id)
-            if active_id and rec.lang_id in active_id.lang_ids:
+            if not rec.lang_id or not backend:
                 rec.is_valid = True
```
Comment (hparfr) [inline shopinvader_search_engine/models/se_index.py:22]: rec.is_valid when no backend ?

File: shopinvader_search_engine/models/shopinvader_backend.py [hunk 3]
```diff
@@ -35,29 +41,68 @@ def clear_index(self):
         return True
 
     def add_missing_index(self):
+        _logger.warning("DEPRECATED: add_missing_index, use action_add_missing_indexes")
+        self.action_add_missing_indexes()
```
Comment (hparfr) [inline shopinvader_search_engine/models/shopinvader_backend.py:45]:
```suggestion
        return self.action_add_missing_indexes()
```

File: shopinvader_search_engine/models/shopinvader_backend.py [hunk 4]
```diff
@@ -35,29 +41,68 @@ def clear_index(self):
...
+
+    def action_add_missing_indexes(self):
         self.ensure_one()
-        ir_models = self._get_default_models()
-        index_obj = self.env["se.index"]
-        ir_export_obj = self.env["ir.exports"]
         if not self.se_backend_id:
             return
-        for model in ir_models:
+        self._add_missing_indexes()
```
Comment (hparfr) [inline shopinvader_search_engine/models/shopinvader_backend.py:51]:
```suggestion
       return self._add_missing_indexes()
```

**Review by hparfr — CHANGES_REQUESTED** [review]
Interesting.

Some minor remarks about code.

File: shopinvader_search_engine/models/shopinvader_backend.py (hunk 4 above)
Comment (simahawk) [reply shopinvader_search_engine/models/shopinvader_backend.py:51]: this is an action called from a btn you should not return records

File: shopinvader_search_engine/models/se_index.py (hunk 2 above)
Comment (simahawk) [reply shopinvader_search_engine/models/se_index.py:22]: sure. If we are not in the context of a backend the index is always valid (See `depends_context`)

File: shopinvader_search_engine/models/se_index.py (hunk 1 above)
Comment (simahawk) [reply shopinvader_search_engine/models/se_index.py:23]: for me is more clear like this: I don't want to to/read anything else after this point, just move on.

**Review by simahawk — COMMENTED** [review]
@hparfr can you update your review? 
I've rebased and dropped the temp dependency.

**Review by hparfr — APPROVED** [review]

**Pull request merged**